src/pybind/mgr/dashboard/frontend/src/app/shared/components/progress/progress.component.html

```diff
@@ -0,0 +1,60 @@
+<div class="d-flex flex-column justify-content-center align-items-center bold">
+  <ng-container>
+    <h3 class="text-center"
+        [ngClass]="{ 'mt-3': inProgress && isPaused }">
+      <i *ngIf="inProgress && isPaused; else spinningIcon"
+         [ngClass]="[icons.large, icons.spinner]"></i>
+      <ng-template #spinningIcon>
+        <i [ngClass]="[icons.large, icons.spin, icons.spinner]"></i>
+      </ng-template>
+    </h3>
+
+    <h3 class="text-center"
+        [ngClass]="inProgress && isPaused ? ['mt-3', 'mb-4'] : 'mt-2'"
+        *ngIf="description">
+      {{ description }}
+    </h3>
+
+    <h5 class="text-center mt-3"
+        *ngIf="currentStep && inProgress && !isPaused">
+      {{ currentStep }}
+    </h5>
+
+    <h5 class="text-center mt-3"
+        *ngIf="executingTaskName">
+      {{ executingTaskName }}
+    </h5>
+  </ng-container>
+
+  <div class="w-50 row h-100 d-flex justify-content-center align-items-center mt-4">
+    <div class="text-center w-75">
+      <ng-container *ngIf="servicesComplete && servicesComplete.length > 0"
+                    i18n>
+          Finished upgrading:
```

Comment:
this is generic so shouldn't we use generic message here?

Reply:
Done